src/components/FileTree/FileTree.tsx

```diff
@@ -5,29 +5,23 @@ import { ReactNode, useState } from 'react'
 
 import { FileList } from './FileList'
 import { css } from '@emotion/react'
-import { useDebounce } from 'react-use'
+import { FileItem } from './types'
 
 interface FileTreeProps {
   label: string
-  files: string[]
-  viewPath: string
+  files: FileItem[]
   noFilesMessage?: string
   actions?: ReactNode
 }
 
 export function FileTree({
   label,
-  files: filesRaw,
-  viewPath,
+  files,
   noFilesMessage = 'No files found',
   actions,
 }: FileTreeProps) {
   const [open, setOpen] = useState(true)
 
-  // Debounce to avoid flickering in of files when renaming
-  const [files, setFiles] = useState<string[]>([])
-  useDebounce(() => setFiles(filesRaw), 50, [filesRaw])
-
```

Comment:
This debounce caused a flicker when clearing the search results. Instead of a debounce, I added an optimistic update of the file name after the `renameFile` call has completed.